### ISSUE TITLE
Mixing up classes in some tests

### ISSUE THREAD

**Issue by DanielBadura** [vimeo/psalm#8357]
Our CI build failed for auto upgrading our dependencies. In there psalm was updated from 4.24.0 to 4.25.0. Psalm now reports some strange errors like:
```
ERROR: UndefinedMethod - tests/Unit/Aggregate/AggregateRootTest.php:58:43 - Method Patchlevel\EventSourcing\Tests\Unit\Aggregate\AggregateRootTest::id does not exist (see https://psalm.dev/022)
        self::assertEquals($id, $profile->id());
```
Assuming that `$profile` is an instance of `AggregateRootTest` but it is instead an instance of `AggregateRoot`. See here for the code: https://github.com/patchlevel/event-sourcing/blob/2.1.x/tests/Unit/Aggregate/AggregateRootTest.php#L54-L58

Also here the full error list from our workflow: https://github.com/patchlevel/event-sourcing/runs/7616545968?check_suite_focus=true

Since the dependency update PR did update multiple deps, i tried it locally with only updating psalm with the same result.

I'm not sure whats going on if this is something on our side or broke with the new release. All errors are located in the tests folder and also only a tiny bit of these tests are affected.

**Comment by AndrolGenhald** [comment]
@someniatko re #8249, care to take a look? Just from skimming through it I think this is a separate case from #8257.

**Comment by someniatko** [comment]
@AndrolGenhald isn't https://github.com/vimeo/psalm/pull/8335 supposed to fix the regression?

**Comment by AndrolGenhald** [comment]
@someniatko Sorry, forgot about that one, @DanielBadura could you test with `4.x-dev`?

**Comment by someniatko** [comment]
IMO 4.25.1 should be released with the regression fix, because it's already the second encounter of the issue.

**Comment by AndrolGenhald** [comment]
Agreed, I've got no idea how to do releases though so I'll just let @orklah handle it :slightly_smiling_face:

**Comment by someniatko** [comment]
p.s. if our suggestion is correct, it should be a duplicate of https://github.com/vimeo/psalm/issues/8330

**Comment by DanielBadura** [comment]
Can confirm that no issue occur with `4.x-dev 6998fab`

**Comment by orklah** [comment]
Released a 4.26.0 :)
